### PR TITLE
`Development`: Remove unused code related to ngx graph

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a [very good performance](https://docs.artemis.cit.tum.de/dev/guidelines/performance/) and prevented too many (unnecessary) and too complex database calls.
- [x] I **removed** multiple integration tests (Spring) related to the features (with a high test coverage).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data (e.g. using paging).
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I **removed** multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).

### Motivation and Context

The nix graph representing competencies and their learning objects was replaced in multiple previous PRs and is unused now.

### Description

This PR removes all legacy learning path code for the server and client.
The removed client code was covered better than the average, so the coverage thresholds have to be lowered a bit.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Course with competencies and exercises

1. Log in to Artemis
2. Check that the learning paths page for instructors or any clickable element in it does not look broken or completely empty
3. Check that the learning path view for a students or any clickable element in it does not look broken or completely empty

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [ ] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

## Summary by CodeRabbit

- **New Features**
	- Removed support for Ngx representations of learning paths, including related components and services.
- **Bug Fixes**
	- Adjusted code coverage thresholds for various metrics.
- **Documentation**
	- Updated routing configuration for learning paths to load components directly instead of modules.
- **Chores**
	- Deleted multiple components, services, and associated tests related to the Ngx learning path functionality.
	- Streamlined the `LearningPathStudentPageComponent` by simplifying route parameter extraction.

### CODE REVIEW

File: src/main/webapp/app/overview/courses-routing.module.ts
Comment (Strohgelaender) [inline src/main/webapp/app/overview/courses-routing.module.ts:194]: Do you want to keep this outcommented?

**Review by cremertim — DISMISSED** [review]
Clean 👍

File: src/main/webapp/app/overview/courses-routing.module.ts
Comment (JohannesWt) [reply src/main/webapp/app/overview/courses-routing.module.ts]: Thanks for the hint, we missed that comment

**Review by iyannsch — DISMISSED** [review]
Tested on TS6, could not find any broken button 👍 I just (not related to this PR) noted that I was able to import competencies from any other course - is this intended? Doesn't this leak potentially confidential data to other course admins?

**Review by cremertim — APPROVED** [review]
Code

**Review by sachmii — APPROVED** [review]
Tested on TS3, could not notice anything broken. Everything wroked fine

**Review by florian-glombik — DISMISSED** [review]
Approve code

The removed files and code seem reasonable, the couple of added lines are fine

**Review by HawKhiem — APPROVED** [review]
Tested on TS1. Everything was great!

**Review by SindiBuklaji — APPROVED** [review]
Tested on Ts1. Everything works as expected👍

**Review by alekspetrov9e — APPROVED** [review]
Tested on TS1. No issues were found. Could generate learning paths and also view learning paths.

**Review by coolchock — APPROVED** [review]
code LGTM

**Review by florian-glombik — APPROVED** [review]
Re-approve code after merge

**Review by sachmii — APPROVED** [review]

**Pull request merged**
